src/scrapers/hapoalim.js

```diff
@@ -142,18 +91,84 @@ function createLoginFields(credentials) {
 }
 
 class HapoalimScraper extends BaseScraperWithBrowser {
+  // eslint-disable-next-line class-methods-use-this
+  get baseUrl() {
+    return 'https://login.bankhapoalim.co.il';
+  }
+
+  // eslint-disable-next-line class-methods-use-this
+  get portalUrl() {
+    return 'rb';
+  }
+
   getLoginOptions(credentials) {
     return {
-      loginUrl: `${BASE_URL}/cgi-bin/poalwwwc?reqName=getLogonPage`,
+      loginUrl: `${this.baseUrl}/cgi-bin/poalwwwc?reqName=getLogonPage`,
       fields: createLoginFields(credentials),
       submitButtonSelector: '#inputSend',
       postAction: async () => waitForRedirect(this.page),
-      possibleResults: getPossibleLoginResults(),
+      possibleResults: this.getPossibleLoginResults(),
     };
   }
 
   async fetchData() {
-    return fetchAccountData(this.page, this.options, (msg) => this.notify(msg));
+    return this.fetchAccountData(this.page, this.options, (msg) => this.notify(msg));
+  }
+
+  async fetchAccountData(page, options) {
```

Comment:
@orzarchi instead of moving those methods into the class, can you move them back and adjust the method signature instead. for example:
```
async fetchAccountData(page, baseUrl, options) {
...
}
```